src/gui/qgsbrowserdockwidget.cpp

```diff
@@ -204,7 +204,8 @@ void QgsBrowserDockWidget::showContextMenu( QPoint pt )
   {
     for ( QMenu *mn : menus )
     {
-      menu->addMenu( mn );
+      QMenu *newMenu = menu->addMenu( mn->title() );  // this method will transfer the ownership
```

Comment:
> // this method will transfer the ownership

I don't think so ;)

Reply:
Maybe I just have mistaken here http://doc.qt.io/qt-5/qmenu.html#addMenu, doesn't it mean will take care of the ownership?

Reply:
The docs there state that `menu->addMenu( mn )` will not take care of ownership.
What is called here is `menu->addMenu( /*QString*/ title )` which keeps ownership of the newly created menu, but does not know about the menu `mn`.